### ISSUE TITLE
[Bug]: Distorted sound in Source games

### ISSUE THREAD

**Issue by excellproj** [Whisky-App/Whisky#445]
### Description

I ran several games: Portal 2, Half-Life 2 and Team Fortress 2, and I noticed heavily distorted sound both in the menus and in the game. It feels like the sound is playing at 0.5x speed. The sound is playing in jerks.

### Steps to reproduce

To install the Steam app in a bottle, download one of the games (Portal 2 or TF2), and run it

### Expected behaviour

I expected normal sound

### Logs
 
```bash
 Whisky Version: 2.0.0
Date: 27.09.2023, 6:28:15 am
macOS Version: Версія 14.0 (23A344)
Bottle Name: Games
Wine Version: 7.7.0
Windows Version: win10
Bottle URL: /Users/MyName/BottleGames/6AF1EFD3-D918-421F-A576-7E683ECD36F6

Arguments: winecfg -v win10 

Launched Wine (9697)
00f0: thread_get_state failed on Apple Silicon - faking zero debug registers
00f8: thread_get_state failed on Apple Silicon - faking zero debug registers
0110:err:module:open_builtin_so_file failed to load .so lib "/Users/MyName/Library/Application Support/com.isaacmarovitz.Whisky/Libraries/Wine/lib/wine/x86_64-unix/wineusb.sys.so"
0110:err:ntoskrnl:ZwLoadDriver failed to create driver L"\\Registry\\Machine\\System\\CurrentControlSet\\Services\\wineusb": c0000142
002c:err:setupapi:SetupDiInstallDevice Failed to start service L"wineusb" for device L"ROOT\\WINE\\WINEUSB", error 1114.
002c:err:setupapi:SetupDiInstallDevice Failed to control service L"wineusb" for device L"ROOT\\WINE\\WINEUSB", error 1062.
wine: configuration in L"/Users/MyName/BottleGames/6AF1EFD3-D918-421F-A576-7E683ECD36F6" has been updated.
Process exited with code 0
```

### What version of Whisky are you using?

2.0.0

### What version of macOS are you using?

Sonoma (macOS 14)

### Issue Language

- [X] Yes my issue is written in English

**Comment by timsayshey** [comment]
Yeah, Steam runs insanely well but when I launch CounterStrike or CS:S, the framerate is unplayable as well. Feels like it just needs some tweaks, or maybe the Proton compatibility package? Enabling DXVK didn't really help much either.

**Comment by excellproj** [comment]
> I can add that TF2 is able to run while the framerate is so low that it's unplayable.

I agree with you, but if we run a test game where the stable FPS is above 60, but the sound issues persist.

**Comment by excellproj** [comment]
> Yeah, Steam runs insanely well but when I launch CounterStrike or CS:S, the framerate is unplayable as well. Feels like it just needs some tweaks, or maybe the Proton compatibility package? Enabling DXVK didn't really help much either.

Interestingly, Portal 2 works fine except for the same sound issues.

**Comment by noma4i** [comment]
BTW I was able to run CS:GO2 with okayish performance in the menus but still unplayable.

**Comment by MrFrankChen** [comment]
I encountered the same issue on game "The Legend of Sword and Fairy 5 Prequel".

**Comment by IsaacMarovitz** [comment]
This is an issue with Source games in general. The sound is heavily tied to rendering performance. I'm not sure if there's an easy solution to this.

**Comment by sjavora** [comment]
This seems to be connected to physics interactions. HL2 will play fine until you cause a bunch of crates to fall at which point it becomes unplayable.

**Comment by wengland02** [comment]
This seems to an issue with games that use DX9 and older. I am experiencing the same issues with NASCAR Racing 2003 Season and LEGO Island. DgVoodoo2 does not help.

**Comment by jeanlain** [comment]
I'm getting poor performance and audio stuttering in other DX9 games like Tome Raider Anniversary (stuttering is even worse than in Source games).

**Comment by martimcfly** [comment]
maybe it can be fixed similar like this cs2 issues see here:  https://www.youtube.com/watch?v=oSFYFIGS-uc

**Comment by apocalypsemeow** [comment]
> This seems to an issue with games that use DX9 and older. I am experiencing the same issues with NASCAR Racing 2003 Season and LEGO Island. DgVoodoo2 does not help.

I'm trying to play Project Diablo II, which may run on DX9 or older since it is a modified Diablo II. Similar sound issues to OP.

**Comment by racinginfinite87** [comment]
What I figured out from one's comment on youtube video on how to install Portal 2 using Whisky, changing your output device's format to 44100 Hz in Audio MIDI Setup helps to solve the issue, at least for Portal 2. Might help with other source games too, although I haven't tested it yet.

**Comment by cgarst** [comment]
Also noticed this with Worms Armageddon, DX9. Alternatively, it's working OK through Wine via Porting Kit.

**Comment by travissanderson** [comment]
@kahlos that worked for me as well, thanks for posting!

**Comment by excellproj** [comment]
> @kahlos that worked for me as well, thanks for posting!
Yes, it worked for the MacBook speaker, but unfortunately, it doesn't apply to the AirPods. 🥲

**Comment by wengland02** [comment]
@kahlos This solution did not work with my Mac Mini M2's built in speakers.

**Comment by Gunmetalblue** [comment]
Have same problem with old games: Space Rangers and Magicka 2. The changing format to 44 hz partly solves the problem but still unplayable :(

**Comment by excellproj** [comment]
> > What I figured out from one's comment on youtube video on how to install Portal 2 using Whisky, changing your output device's format to 44100 Hz in Audio MIDI Setup helps to solve the issue, at least for Portal 2. Might help with other source games too, although I haven't tested it yet.
> 
> This pointed me in the right direction for a non-source game. I've found the audio works no problem when using an external audio device, but has constant skipping / judderyness when using the in-built speakers on my M1 MacBook Pro.
> 
> I tried setting the default audio drivers (ALSA, OSS, PulseAudio) in Winetricks settings, but predictably nothing worked besides the default Mac CoreAudio driver.
> 
> Some old wine forum posts lead me to my solution; changing the sample rate in the format settings in the **Audio MIDI Setup** utility: <img alt="image" width="799" src="https://private-user-images.githubusercontent.com/763716/287515465-a41ce7e7-a8e6-459f-98ef-2d1df4cf38f6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQ1ODE5MzEsIm5iZiI6MTcwNDU4MTYzMSwicGF0aCI6Ii83NjM3MTYvMjg3NTE1NDY1LWE0MWNlN2U3LWE4ZTYtNDU5Zi05OGVmLTJkMWRmNGNmMzhmNi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMTA2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDEwNlQyMjUzNTFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04ZTI1NjYzOTM3MTJkODU0ODIwMTUzNjAxZDJhZTI2NmI3N2IyYjZmNWY2MWMyMTM2MmQ0MmNkODdiYzRkZGE3JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.WA07upwbXBIzNzRp6MgpV-YDvFcNdHKB6_-SCqeNXuw"> <img alt="image" width="279" src="https://private-user-images.githubusercontent.com/763716/287515510-4ba810b1-9d04-426e-b030-7cfc1caa79be.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQ1ODE5MzEsIm5iZiI6MTcwNDU4MTYzMSwicGF0aCI6Ii83NjM3MTYvMjg3NTE1NTEwLTRiYTgxMGIxLTlkMDQtNDI2ZS1iMDMwLTdjZmMxY2FhNzliZS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMTA2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDEwNlQyMjUzNTFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0yMmQ5MGVmMGY2NzdiNTU5YmIwYzg5NWRmOTlmZTYwY2IyZWJmNjQyYmE0NTgyY2ZiZjI0YTQyZGQ4ODg0YjkxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.FvIKmuWwcp890_4y7j0GxUCJ6LKYQ6oUcwldawPQf6s">
> 
> Adjusting to 44.1 kHz resolved the skipping / juddery issues for me on the built-in M1 MacBook Pro speakers. A little odd, as nothing is running through MIDI per se, but I guess Wine's CoreAudio driver pulls its sample rate through from this setting? Either way, worth a try if you're getting audio issues

I tried these settings on the game Portal: Revolution, but unfortunately, they didn't work.

**Comment by roryabraham** [comment]
> Some old wine forum posts lead me to my solution; changing the sample rate in the format settings in the Audio MIDI Setup utility

This seems to have completely fixed the audio for me on Age of Mythology Extended Edition (2022 M2 MacBook Pro, using wired headphones)

**Comment by MaxGaming64** [comment]
I managed to fix it by setting the speaker configuration to 2 speakers (which is what I have), from the default setting (7.1 speakers). You may need to set the speaker configuration to your amount.

**Comment by undakovr** [comment]
> > What I figured out from one's comment on youtube video on how to install Portal 2 using Whisky, changing your output device's format to 44100 Hz in Audio MIDI Setup helps to solve the issue, at least for Portal 2. Might help with other source games too, although I haven't tested it yet.
> 
> This pointed me in the right direction for a non-source game. I've found the audio works no problem when using an external audio device, but has constant skipping / judderyness when using the in-built speakers on my M1 MacBook Pro.
> 
> I tried setting the default audio drivers (ALSA, OSS, PulseAudio) in Winetricks settings, but predictably nothing worked besides the default Mac CoreAudio driver.
> 
> Some old wine forum posts lead me to my solution; changing the sample rate in the format settings in the **Audio MIDI Setup** utility: <img alt="image" width="799" src="https://private-user-images.githubusercontent.com/763716/287515465-a41ce7e7-a8e6-459f-98ef-2d1df4cf38f6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTk0NTkzNjksIm5iZiI6MTcxOTQ1OTA2OSwicGF0aCI6Ii83NjM3MTYvMjg3NTE1NDY1LWE0MWNlN2U3LWE4ZTYtNDU5Zi05OGVmLTJkMWRmNGNmMzhmNi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNjI3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDYyN1QwMzMxMDlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wYzAyYjBhYWJjNDgwYjM0MDc1YmI0YjRlMzkyMzk1ZmQ2ZDgxODAzMzM3ODRiNjUxZDZiYjFjOWExZWY3OTJlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.JdaD-PHgASzVId0AjJKDTlY3cgq2EDm8hc_GyEW16LY"> <img alt="image" width="279" src="https://private-user-images.githubusercontent.com/763716/287515510-4ba810b1-9d04-426e-b030-7cfc1caa79be.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTk0NTkzNjksIm5iZiI6MTcxOTQ1OTA2OSwicGF0aCI6Ii83NjM3MTYvMjg3NTE1NTEwLTRiYTgxMGIxLTlkMDQtNDI2ZS1iMDMwLTdjZmMxY2FhNzliZS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNjI3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDYyN1QwMzMxMDlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wMWIzYjY0ZjdlMWJhNTE2OGJmYzNhMGE5MTA3ZjJkZjg1ZWE1NjIwOTFlYWJjMDcwMWUwM2M1ZDZlZDg3YmI5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.ARxMaBHiN0SlXpPzmNCMop-1FxyMU5DzIQWclZeJFtA">
> 
> Adjusting to 44.1 kHz resolved the skipping / juddery issues for me on the built-in M1 MacBook Pro speakers. A little odd, as nothing is running through MIDI per se, but I guess Wine's CoreAudio driver pulls its sample rate through from this setting? Either way, worth a try if you're getting audio issues

thanks. changing the the hz to 4100 made it work flawlessly

**Comment by AliceMousie** [comment]
I was having this issue in CS2 despite my headphones already being at 44.1KHz with no option to change them. Fix for me was adding `-directsound` to the launch options. 

I also seem to have a lower performance floor when setting `-threads 8` in launch options, especially in gunfights and in offline with bots, but this might have to be a lower number if your CPU doesn't have enough cores.

**Comment by alimyuz** [comment]
Adding `-directsound` in Deadlock launch option completely disabled the sound for me. 44.1KHz option also did not work.

**Comment by ywake** [comment]
I also faced the same problem and could not solve it with MIDI settings, but I solved it by updating `wine` to the latest version. 

Here's how I updated wine.
```bash
# install wine via homebrew
brew install --cask --no-quarantine wine-stable

# Check the location of wine
where wine64
# output:
# /Users/your_name/Library/Application Support/com.isaacmarovitz.Whisky/Libraries/Wine/bin/wine64
# /opt/homebrew/bin/wine64

# Remove the old wine
rm /Users/your_name/Library/Application\ Support/com.isaacmarovitz.Whisky/Libraries/Wine/bin/wine64

# Create a symbolic link to the new wine
ln -s /opt/homebrew/bin/wine64 /Users/your_name/Library/Application\ Support/com.isaacmarovitz.Whisky/Libraries/Wine/bin/wine64
```

After that, the problem was resolved by recreating the bottle.

I actually did it in a different way, so there may be some errors, but I hope it helps.

**Comment by dr-dimitru** [comment]
I experienced the same switching to Mono inside the game solved issue for me; But it's terrible for shooters

**Comment by SlavenIvanov** [comment]
I had the audio issue with deadlock.
Tried 44 100HZ and it didn't work.
But changing it to 96 000Hz did the trick!
<img width="294" alt="Screenshot 2024-08-25 at 20 23 44" src="https://github.com/user-attachments/assets/084da39e-1f56-4140-9c4b-dd544a995984">
